util/movement-algs/src/grouping_heuristic/chunking.rs

```diff
@@ -0,0 +1,72 @@
+use crate::grouping_heuristic::{GroupingHeuristic, GroupingOutcome};
+
+pub struct Chunking {
+	pub size: usize,
+}
+
+impl Chunking {
+	pub fn new(size: usize) -> Self {
+		Self { size }
+	}
+
+	pub fn boxed(size: usize) -> Box<Self> {
+		Box::new(Self::new(size))
+	}
+}
+
+impl<T> GroupingHeuristic<T> for Chunking {
+	fn distribute(
+		&mut self,
+		distribution: Vec<GroupingOutcome<T>>,
+	) -> Result<Vec<GroupingOutcome<T>>, anyhow::Error> {
+		// flatten the distribution
+		let mut distribution = distribution
+			.into_iter()
+			.flat_map(|outcome| outcome.into_inner())
+			.collect::<Vec<_>>();
+
+		// chunk the distribution
+		let mut chunks = Vec::new();
+		while !distribution.is_empty() {
+			let chunk = distribution.drain(0..self.size.min(distribution.len())).collect();
```

Comment:
It needs to take ownership and this was before I added `itertools`.  Also, this is quadratic because of the resize of the vector no? Should I just use `VecDeque`, so that I can pop off the front?

Reply:
Does `itertools` take care of this internally...

Reply:
Seems like their `GroupBy` does.